qark/plugins/Dynamically_Loading_Code.py

```diff
@@ -1,66 +1,121 @@
-import sys, os, re
+import sys
+import os
+import re
 sys.path.insert(0, os.path.dirname(os.path.realpath(__file__)) + '../lib')
 from yapsy.IPlugin import IPlugin
 from plugins import PluginUtil
 from modules import common
 from lib.pubsub import pub
 import lib.plyj.parser as plyj
+import lib.plyj.model as m
 
 
 class DynamicallyLoadingCodePlugin(IPlugin):
     DEX_CLASS_LOADER = r'DexClassLoader'
     CLASS_LOADER = r'loadClass'
     DYNAMIC_BROADCAST_RECEIVER = r'registerReceiver'
+    global receivers_list
+    receivers_list = []
 
     def __init__(self):
         self.name = 'Dynamically loading code'
 
+    # recursive function to find the loadclass function name by traversing down the AST
+    def recursive_classloader_function(self, fields, file, res):
+        if type(fields) is m.MethodDeclaration:
+            if str(fields.name) == self.CLASS_LOADER:
+                PluginUtil.reportInfo(file_name, self.class_loader_issue(file_name), res)
+            elif self.CLASS_LOADER in str(fields):
+                PluginUtil.reportInfo(file_name, self.class_loader_issue(file_name), res)
+        elif type(fields) is list:
+            for tree_object_fields in fields:
+                self.recursive_classloader_function(tree_object_fields, file, res)
+        elif hasattr(fields, '_fields'):
+            for values in fields._fields:
+                self.recursive_classloader_function(getattr(fields, values), file, res)
+        return
+
+    # recursive function to find the register receiver function name by traversing down the AST
+    def recursive_register_receiver_function(self, fields, file, res):
+        if type(fields) is m.MethodDeclaration:
+            if str(fields.name) == self.DYNAMIC_BROADCAST_RECEIVER:
+                receivers_list.append(file_name)
+            elif self.DYNAMIC_BROADCAST_RECEIVER in str(fields):
+                if file_name not in receivers_list:
+                    receivers_list.append(file_name)
+        elif type(fields) is list:
+            for tree_object_fields in fields:
+                self.recursive_register_receiver_function(tree_object_fields, file, res)
+        elif hasattr(fields, '_fields'):
+            for values in fields._fields:
+                self.recursive_register_receiver_function(getattr(fields, values), file, res)
+        return
+
     def target(self, queue):
         files = common.java_files
-        global parser, tree, fileName
+        global parser, tree, file_name
         parser = plyj.Parser()
         tree = ''
         res = []
-
-        #List of Broadcast Receiver
-        list_BR = []
         count = 0
-        for f in files:
+        for file in files:
```

Comment:
`f` is fine here. If you use `file` as a variable it shadows the python `file` type.